docs/host-powershell/README.md

```diff
@@ -0,0 +1,149 @@
+# Host PowerShell Core in .NET Core Applications
+
+> This documentation is based on PowerShell Core packages built against .NET Core 1.1 and prior. 
+> Things may change after we move to .NET Core 2.0.
+
+## PowerShell Core targeting .NET Core 1.1 and Prior
+
+### Overview
+
+Due to the lack of necessary APIs for manipulating assemblies in .NET Core 1.1 and prior,
+PowerShell Core needs to control assembly loading via our customized `AssemblyLoadContext` in order to do tasks like type resolution.
+So applications that want to host PowerShell Core (using PowerShell APIs) need to be bootstrapped from `PowerShellAssemblyLoadContextInitializer`.
+
+`PowerShellAssemblyLoadContextInitializer` exposes 2 APIs for this purpose:
+`SetPowerShellAssemblyLoadContext` and `InitializeAndCallEntryMethod`.
+They are for different scenarios:
+
+- For `SetPowerShellAssemblyLoadContext`,
+it's designed to be used by a native host whose TPA (Trusted Platform Assemblies) list doesn't include powershell assemblies,
+such as the in-box `powershell.exe` and other native CoreCLR host in Nano Server.
+When using this API, instead of setting up a new load context,
+`PowerShellAssemblyLoadContextInitializer` will register a handler to the [Resolving][] event of the default load context.
+Then PowerShell Core will depend on the default load context to handle TPA and the `Resolving` event to handle other assemblies.
+
+- For `InitializeAndCallEntryMethod`,
+it's designed to be used with `dotnet.exe` where the TPA list includes powershell assemblies.
+When using this API, `PowerShellAssemblyLoadContextInitializer` will set up a new load context to handle all assemblies.
+PowerShell Core itself also uses this API for [bootstrapping][].
+
+This documentation only covers the `InitializeAndCallEntryMethod` API,
+as it's what you need when building a .NET Core application with .NET CLI.
+
+### Example
```

Comment:
There should be three sub-headings here:
1. Hosting Windows PowerShell
2. Hosting PowerShell in .Net Core
3. Example .Net Core program